### DISCUSSION TITLE
New restart methods?

### DISCUSSION THREAD

**Discussion by kibernetik542** [kubernetes-client/csharp#867] [Q&A]
I have logic for restarting deployment, daemon set and stateful set via using patch.

I have 2 suggestion

1 - to create new ready methods for Restart(...)
2 - in next release to use dependency for Json.Patch ( because of System.Text.Json not compatible with Mircrosoft.AspNetCore.JsonPatch )

as it is still in open on [aspnet core github issue](https://github.com/dotnet/aspnetcore/issues/24333) you think it will be better to wait for their support or we can use  Json.Patch as dependency?

I can make PR on examples if you are ok with ideas.

**Comment by tg123** [comment]
1 see https://github.com/kubernetes-client/python/blob/c66a39108b06091417cdbcd408d6e3d037b446fe/examples/deployment_crud.py#L107

the plan for c# sdk is to create a kubectl experience api. the subproject is starting soon, PRs are welcomed

2 not quite get ` next release to use dependency for Json.Patch `
the sdk does not have any dependencies to aspnetcore nor json.patch. the reason see #521. the obj can be generated from any library.


the example of patch in [ `examples/patch`](https://github.com/kubernetes-client/csharp/blob/master/examples/patch/Program.cs) illustrate how to use merge patch

and e2e tests depends on json.patch
https://github.com/kubernetes-client/csharp/blob/92ccac423a4e9eacd058aceb00bed02150b54cb3/tests/E2E.Tests/MnikubeTests.cs#L74

**Comment by kibernetik542** [comment]
> 1 see https://github.com/kubernetes-client/python/blob/c66a39108b06091417cdbcd408d6e3d037b446fe/examples/deployment_crud.py#L107
> 
> the plan for c# sdk is to create a kubectl experience api. the subproject is starting soon, PRs are welcomed
> 
> 2 not quite get `next release to use dependency for Json.Patch` the sdk does not have any dependencies to aspnetcore nor json.patch. the reason see #521. the obj can be generated from any library.
> 
> the example of patch in [ `examples/patch`](https://github.com/kubernetes-client/csharp/blob/master/examples/patch/Program.cs) illustrate how to use merge patch
> 
> and e2e tests depends on json.patch
> 
> https://github.com/kubernetes-client/csharp/blob/92ccac423a4e9eacd058aceb00bed02150b54cb3/tests/E2E.Tests/MnikubeTests.cs#L74

well about 2nd point I meant `new V1Patch(patch, V1Patch.PatchType.JsonPatch)` this option not working with [according to this issue](https://github.com/dotnet/aspnetcore/issues/24333)

about python approach, I tried also restartedAt option but I can assure it changes ( edits ) yaml file but not restarts pods. I have different solution for this which works for all. My idea is instead of all time creating Patch just to create direct restart methods. I will make PR for restart options. Thanks for reply
